shop/src/main/java/ro/msg/learning/shop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package ro.msg.learning.shop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.StockDto;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.OrderDetailRepository;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.EmailService;
+import ro.msg.learning.shop.service.OrderService;
+import ro.msg.learning.shop.service.strategy.OrderStrategy;
+import ro.msg.learning.shop.utils.mapper.OrderMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
```

Comment:
The repository of one entity should communicate only with the service of the respective entity. This would mean that here you should have injected the services of the entities instead of the repositories. This would also mean that the services would need the functions that would be used here